### DISCUSSION TITLE
replication lag is too big

### DISCUSSION THREAD

**Discussion by iushas** [apache/kvrocks#1504] [Q&A]
redis:
<img width="703" alt="image" src="https://github.com/apache/incubator-kvrocks/assets/7236171/b1ae2cee-7ed8-4b98-ad74-9c9b1318d428">

kvrocks:
<img width="665" alt="image" src="https://github.com/apache/incubator-kvrocks/assets/7236171/6f035e6c-c33f-49bb-a9bb-e19bf5e1534e">

use the same press test command， redis's master  replication lag is always 0 or 1 ， kvrocks‘s lag is very big, why?     
this can lead slave after master too much， how to resove!!

**Comment by git-hulk** [comment]
The lag is totally different between Kvrocks and Redis. For the Redis, the lag means the replication seconds between the master and replica. But for the Kvrocks, it represents the write sequence gap. So I think you can relax the alert threshold value for the Kvrocks' lag like 1024 or others.

**Reply by iushas** [reply]
sorry ， 
did not understand.  What's the write sequence gap mean， could  you tell me more about it？

**Reply by git-hulk** [reply]
Sure, the write sequence is equal to the number of KV that were written in RocksDB. For example, it will increase 1 if the command is `set a b`, and increase 2 if the command is `hset hash f1 v1` since it will write the metadata and subkey.

**Reply by iushas** [reply]
then how we know kvrocks  slave and master's offset diff ,  like redis ？

**Reply by git-hulk** [reply]
For the Kvrocks, the lag is the indeed replication offset, but for the Redis is the last replication/interaction time. So as mentioned in the first comment, you should relax the alert threshold value for Kvrocks.

**Reply by iushas** [reply]
ok ，understand, thanks !
